qiita_plugins/target_gene_type/target_gene_type/summary.py

```diff
@@ -103,35 +103,37 @@ def _summary_not_demultiplexed(artifact_type, filepaths):
     """
     # loop over each of the fps/fps_type pairs
     artifact_information = []
-    for (fps, fps_type) in filepaths:
-        # Step 2: generate HTML summary
-        # md5, from http://stackoverflow.com/a/3431838
-        with open(fps, "rb") as f:
-            hash_md5 = md5()
-            for chunk in iter(lambda: f.read(4096), b""):
-                hash_md5.update(chunk)
-
-        # getting head of the files
-        header = []
-        if artifact_type not in FILEPATH_TYPE_TO_NOT_SHOW_HEAD:
-            # we need to encapsulate the full for loop because gzip will not
-            # raise an error until you try to read
-            try:
-                with gopen(fps, 'r') as fin:
-                    header = [
-                        next(fin) for x in xrange(LINES_TO_READ_FOR_HEAD)]
-            except IOError:
-                with open(fps, 'r') as fin:
-                    header = [
-                        next(fin) for x in xrange(LINES_TO_READ_FOR_HEAD)]
-        filename = basename(fps)
-        artifact_information.append("<h3>%s (%s)</h3>" % (filename, fps_type))
-        artifact_information.append("<b>MD5:</b>: %s</br>" %
-                                    hash_md5.hexdigest())
-        if header:
-            artifact_information.append(
-                "<p style=\"font-family:'Courier New', Courier, monospace;"
-                "font-size:10;\">%s</p><hr/>" % ("<br/>".join(header)))
+    for (fps_type, fps) in sorted(filepaths.items()):
+        for fp in fps:
+            # Step 2: generate HTML summary
+            # md5, from http://stackoverflow.com/a/3431838
+            with open(fp, "rb") as f:
+                hash_md5 = md5()
+                for chunk in iter(lambda: f.read(4096), b""):
+                    hash_md5.update(chunk)
+
+            # getting head of the files
+            header = []
+            if artifact_type not in FILEPATH_TYPE_TO_NOT_SHOW_HEAD:
+                # we need to encapsulate the full for loop because gzip will
+                # not raise an error until you try to read
+                try:
```

Comment:
Instead of doing a `try/except` you could read in the magic number right? It would probably amount to the same lines of code in total, but at least there wouldn't be an except `IOError` (which can also be raised if the file doesn't exist IIRC).